src/discof/repair/fd_repair_tile.c

```diff
@@ -358,18 +430,86 @@ fd_repair_sign_and_send( fd_repair_tile_ctx_t *  repair_tile_ctx,
      ^                ^
      buf              buf+4 */
 
-  fd_signature_t sig;
-  repair_signer( repair_tile_ctx, sig.uc, buf, buflen, FD_KEYGUARD_SIGN_TYPE_ED25519 );
+  /* If async, we send the signing request to the sign tile */
+  if( is_async ) {
+    repair_signer_async( repair_tile_ctx, nonce, buf, buflen, FD_KEYGUARD_SIGN_TYPE_ED25519, sign_out);
+    return buflen + 64UL;
+  /* If sync, we sign using keyguard */
+  } else {
+    fd_signature_t sig;
+    repair_signer( repair_tile_ctx, sig.uc, buf, buflen, FD_KEYGUARD_SIGN_TYPE_ED25519 );
+
+    /* Reintroduce the signature */
+    buf    -= 64UL;
+    buflen += 64UL;
+    fd_memcpy( buf + 4U, &sig, 64U );
 
-  /* Reintroduce the signature */
+    return buflen;
+  }
+}
 
-  buf    -= 64UL;
-  buflen += 64UL;
-  fd_memcpy( buf + 4U, &sig, 64U );
+/* Returns a sign_out context that has available credits.
+   If no sign_out context has available credits, returns NULL. */
+fd_repair_out_ctx_t *
+fd_sign_avail_credits( fd_repair_tile_ctx_t * ctx ) {
```

Comment:
make this static, don't prefix with `fd` (fd_sign implies it's part of the sign module)